### PR TITLE
Payment routes db integration

### LINKED ISSUE #58: Payment routes - integrate Stripe WIPs with DB
- [x] Save new payment methods, customers,  and subscriptions to DB.
- [x] Save changes to payment methods, customers,  and subscriptions in DB.
- [x] Prevent plans switches that fail maximum user requirements at that pricing band.
- [x] Add documentation across paymentRoutes.

### PR DESCRIPTION
**Pull request by mark-marshall**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

1) Adds DB saving functionality in-line with new Hotel schema and the billing sub-doc.
2) Documentation for paymentRoutes.

Fixes #58

## Change status

- [ ] Complete, tested, ready to review and merge

- [x] Complete, but not tested (may need new tests)

- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

Via react-stripe-elements, Compass, and Stripe Dashboard. Formal testing of this is on-hold due to the difficult nature of testing Stripe functionality and prioritisation of other issues on the project.

# Checklist:

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have made corresponding changes to the documentation, if relevant

- [x] I have run the app using my feature and ensured that no functionality is broken

- [x] My changes generate no new warnings

- [x] There are no merge conflicts

### CODE REVIEW

File: utils/path.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const path = {
   auth: '/api/auth',
   hotel: '/api/hotel',
   subscribe: '/api/subscription',
+  getStaff: 'https://web-chat-labs.herokuapp.com/api/users?hotel_id=',
 };
```
Comment (Pav0l) [inline utils/path.js]: A suggestion - move the URL `https://web-chat-labs.herokuapp.com` to a separate `url.js` file inside utils folder and use that and path for `/api/users` with query string from `path.js`. The reasoning is to keep separated concerns in separated files (url, path).

File: utils/errorMessage.js [hunk 2]
```diff
@@ -43,6 +43,9 @@ const errorMessage = {
   invalidPlan: {
     message: 'The plan you requested does not exist.',
   },
+  tooManyUsers: {
+    message: 'There are too many staff accounts on this hotel to switch plans.',
```
Comment (Pav0l) [inline utils/errorMessage.js]: I'd rephrase this so it's more descriptive. Something like: 
`The number of staff accounts on this hotel is too high for selected plan.`

**Review by Pav0l — CHANGES_REQUESTED** [review]
Great job on Stripe integration @mark-marshall ! 🥇 

These comments below are not change request per se, but rather my suggestions/opinions.

**Review by Velsu — APPROVED** [review]
Amazing job Mark

**Review by Pav0l — APPROVED** [review]
